File: tests/factories/listing.py
import factory
import random

from salsa import models

from .base import ModelFactory
from .user_account import UserAccountFactory
from .product import ProductFactory


class ListingFactory(ModelFactory):
    active = True
    name = factory.Faker('text')
    description = factory.Faker('text')
    price = random.randint(1,101)
    amount_available = random.randint(1,101)

    product = factory.SubFactory(ProductFactory)
    user_account = factory.SubFactory(UserAccountFactory)

    class Meta:
        model = models.Listing

File: flutter/helistrong/lib/models/listing.dart
import 'package:helistrong/models/product.dart';
import 'package:helistrong/models/user.dart';

class Listing {
  final String id;
  final bool active;
  final String name;
  final double price;
  final DateTime updatedAt;
  final String description;
  final DateTime createdAt;
  final double amountAvailable;
  final String productId;
  final User user;
  final Product product;

  Listing(this.id, this.active, this.name, this.price, this.updatedAt, this.description, this.createdAt, this.amountAvailable, this.productId, this.user, this.product);
}